src/Umbraco.Web.UI.Client/src/views/propertyeditors/datepicker/datepicker.controller.js

```diff
@@ -1,30 +1,84 @@
+
+    $scope.model.config.calendarWeeks = $scope.model.config.calendarWeeks == 0 ? false : true;
+
+    if ($scope.model.config.daysOfWeekDisabled !== undefined || $scope.model.config.daysOfWeekDisabled !== null) {
+        $scope.model.config.daysOfWeekDisabled = angular.isArray($scope.model.config.daysOfWeekDisabled) ? $scope.model.config.daysOfWeekDisabled : [];
+    }
+
+    if ($scope.model.config.minuteStepping !== undefined || $scope.model.config.minuteStepping !== null) {
+        $scope.model.config.minuteStepping = parseInt($scope.model.config.minuteStepping) > 0 ? parseInt($scope.model.config.minuteStepping) : 1;
+    }
+
+    if ($scope.model.config.minDate !== "" || $scope.model.config.minDate !== undefined || $scope.model.config.minDate !== null) {
+        $scope.model.config.minDate = new Date($scope.model.config.minDate);
+    }
+
+    if ($scope.model.config.maxDate !== "" || $scope.model.config.maxDate !== undefined || $scope.model.config.maxDate !== null) {
+        $scope.model.config.maxDate = new Date($scope.model.config.maxDate);
+    }
+
     //ensure the format doesn't get overwritten with an empty string
     if ($scope.model.config.format === "" || $scope.model.config.format === undefined || $scope.model.config.format === null) {
         $scope.model.config.format = $scope.model.config.pickTime ? "YYYY-MM-DD HH:mm:ss" : "YYYY-MM-DD";
     }
 
+    //set value of these boolean properties based on config format - in v4 and v5 of the datepicker plugin it decide the "viewModes" based on format
+    $scope.model.config.useSeconds = containsChar($scope.model.config.format, "s") ? true : false;
+    $scope.model.config.useMinutes = containsChar($scope.model.config.format, "m") ? true : false;
+    $scope.model.config.pickTime = containsChar($scope.model.config.format, "H") || containsChar($scope.model.config.format, "m") ? true : false;
+    $scope.model.config.pickDate = containsChar($scope.model.config.format, "Y") || containsChar($scope.model.config.format, "M") || containsChar($scope.model.config.format, "D") ? true : false;
+
     $scope.hasDatetimePickerValue = $scope.model.value ? true : false;
     $scope.datetimePickerValue = null;
 
+    $scope.onlyTimePicker = $scope.model.config.pickTime && !$scope.model.config.pickDate;
+    $scope.datetimePickerIcon = $scope.onlyTimePicker ? "time" : "calendar";
+    $scope.errorMsg = {
+        required: "Required",
+        invalid: "Invalid " + ($scope.onlyTimePicker ? "time" : "date")
+    };
+
+    localizationService.localize("general_required").then(function (value) {
+        $scope.errorMsg.required = value;
+    });
+
+    if ($scope.onlyTimePicker) {
+        localizationService.localize("content_invalidTime").then(function (value) {
+            $scope.errorMsg.invalid = value;
+        });
+    }
+    else {
+        localizationService.localize("content_invalidDate").then(function (value) {
+            $scope.errorMsg.invalid = value;
+        });
+    }

```

Comment:
The error messages wasn't localized .. maybe we can use a similar approach for the other property editors?